selfservice/flow/settings/handler.go

```diff
@@ -377,9 +377,16 @@ func (h *Handler) fetchFlow(w http.ResponseWriter, r *http.Request) error {
 
 	if pr.ExpiresAt.Before(time.Now().UTC()) {
 		if pr.Type == flow.TypeBrowser {
+			redirectURL := urlx.AppendPaths(h.d.Config(r.Context()).SelfPublicURL(), RouteInitBrowserFlow)
+			if pr.ReturnTo != "" {
+				query := redirectURL.Query()
+				query.Set("return_to", pr.ReturnTo)
+				redirectURL.RawQuery = query.Encode()
+			}
```

Comment:
This is repetitive, you could instead create a function which generates the detaiols to pass to the error below. That way you can just reuse this. Or, alternative, create an error creation function which you use and pass the config and returnTo as args

Reply:
Updated